data/s3dis/indoor3d_util.py

```diff
@@ -0,0 +1,690 @@
+import glob
+import numpy as np
+import os
+
+# -----------------------------------------------------------------------------
+# CONSTANTS
+# -----------------------------------------------------------------------------
+
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+g_classes = [
+    x.rstrip()
+    for x in open(os.path.join(BASE_DIR, 'meta_data/class_names.txt'))
+]
+g_class2label = {cls: i for i, cls in enumerate(g_classes)}
+g_class2color = {
+    'ceiling': [0, 255, 0],
+    'floor': [0, 0, 255],
+    'wall': [0, 255, 255],
+    'beam': [255, 255, 0],
+    'column': [255, 0, 255],
+    'window': [100, 100, 255],
+    'door': [200, 200, 100],
+    'table': [170, 120, 200],
+    'chair': [255, 0, 0],
+    'sofa': [200, 100, 100],
+    'bookcase': [10, 200, 100],
+    'board': [200, 200, 200],
+    'clutter': [50, 50, 50]
+}
+g_easy_view_labels = [7, 8, 9, 10, 11, 1]
+g_label2color = {g_classes.index(cls): g_class2color[cls] for cls in g_classes}
+
+# -----------------------------------------------------------------------------
+# CONVERT ORIGINAL DATA TO POINTS, SEM_LABEL AND INS_LABEL FILES
+# -----------------------------------------------------------------------------
+
+
+def export(anno_path, out_filename):
+    """Convert original dataset files to points, instance mask and semantic
+    mask files. We aggregated all the points from each instance in the room.
+
+    Args:
+        anno_path: path to annotations. e.g. Area_1/office_2/Annotations/
+        out_filename: path to save collected points and labels
+        file_format: txt or numpy, determines what file format to save.
+    Returns:
+        None
+    Note:
+        the points are shifted before save, the most negative point is now
+            at origin.
+    """
+    points_list = []
+    ins_idx = 1  # instance ids should be 1-indexed, 0 is unannotated
```

Comment:
1-indexed or index-1?

Reply:
The inst ids should be indexed from 1 (1,2,...,inst_num), because 0 is unannotated.

Reply:
Oh I see. It seems a little ambiguous, maybe you can use "1~indexed", or "instance ids should start from 1".

Reply:
Sure.